client/app/bundles/course/survey/containers/ResponseForm/index.jsx

```diff
@@ -14,75 +15,58 @@ const styles = {
 };
 
 const responseFormTranslations = defineMessages({
-  selectAtLeast: {
-    id: 'course.surveys.ResponseForm.selectAtLeast',
-    defaultMessage: 'Please select at least {count} option(s).',
-  },
-  selectAtMost: {
-    id: 'course.surveys.ResponseForm.selectAtMost',
-    defaultMessage: 'Please select at most {count} option(s).',
-  },
   submitted: {
     id: 'course.surveys.ResponseForm.submitted',
     defaultMessage: 'Submitted',
   },
 });
 
-const validateTextResponseAnswer = (formatMessage, answer) => {
-  if (answer.question.required && !answer.text_response) {
-    return { text_response: formatMessage(formTranslations.required) };
-  }
-  return {};
-};
+/**
+ * Merges response answers into survey data to form initialValues for redux-form.
+ * For multiple response questions, selected_options is converted to a format that mirrors checkboxes:
+ * e.g. [121, 123] becomes
+ *      [{ id: 121, selected: true }, { id: 122, selected: false }, { id: 123, selected: true }]
+ */
+export const buildInitialValues = (survey, response) => {
```

Comment:
FYI for submission edit page we are storing the selected_options in a single redux-form field for a group of checkboxes:

https://github.com/coursemology-collab/coursemology2/blob/react/submission-page/client/app/bundles/course/assessment/submission/components/CheckboxFormGroup.jsx